app/src/main/java/com/eveningoutpost/dexdrip/Services/DexCollectionService.java

```diff
@@ -234,6 +235,12 @@ private synchronized void handleConnectedStateChange() {
             Log.d(TAG, "Services already discovered");
             checkImmediateSend();
         }
+
+        if (mBluetoothGatt == null) {
```

Comment:
What is the logic behind this change? When do you expect this condition to be met?

Reply:
I saw a lot time that with blucon really often I don"t get any new data because xdrip thinks blucon is connected but in fact mBluetoothGatt is null (that is the reason why I expended system status) so xdrip doesn't try to reconnect and blucon can't send new data. This is more a workaround, I did not find how this can happens but it definitly happens.

So I assume, either it never happens and code doesn't break anything or it happens and xdrip gets quicker reconnected.

What do you think?